pkg/pgmodel/ingestor/ingestor.go

```diff
@@ -93,19 +93,26 @@ func (ingestor *DBIngestor) Ingest(ctx context.Context, r *prompb.WriteRequest)
 	var (
 		timeseries = r.Timeseries
 		metadata   = r.Metadata
+		size       = r.Size()
 	)
 	release := func() { FinishWriteRequest(r) }
 	switch numTs, numMeta := len(timeseries), len(metadata); {
 	case numTs > 0 && numMeta == 0:
 		// Write request contains only time-series.
 		n, err := ingestor.ingestTimeseries(ctx, timeseries, release)
+		if err == nil {
```

Comment:
Can we defer the error to avoid the repetition?

Reply:
I wanted but due to concurrent nature (below) it's not really straight forward so I skipped it